kotlin/src/main/kotlin/necklace/juwellery/Juwellery.kt

```diff
@@ -0,0 +1,40 @@
+package necklace.juwellery
+
+import necklace.enums.EarringType
+import necklace.enums.Juwel
+import necklace.enums.NecklaseType
+
+
+sealed class Juwellery(open val stone: Juwel) {
+    open val isSmall = false
+    open val isLarge = false
+}
+
+data class Earring(val type: EarringType, override val stone: Juwel) : Juwellery(stone) {
+    override val isSmall: Boolean
+        get() = type == EarringType.Stud
+}
+
+data class Pendant(override val stone: Juwel) : Juwellery(stone) {
+    override val isSmall: Boolean
+        get() = true
+}
+
+data class Ring(override val stone: Juwel): Juwellery(stone)
+
+sealed class Necklace(open val type: NecklaseType, override val stone: Juwel): Juwellery(stone)
```

Comment:
In Kotlin, data classes are the counterpart of C# records (I guess).

Kotlin does not support inheritance for data classes, see e.g. <https://kotlinlang.org/docs/data-classes.html>
Therefor I introduced this intermediate sealed class with the two derived classes `SimpleNecklace` and `PendantNecklace`.

Please note: The property `chain` of a `PendantNecklace` is of type `SimpleNecklace`. Is this correct?

Reply:
it could be a chain or a long chain

Reply:
Hmm, OK. I have to think about this a bit if we could model that in a way such that [illegal states are unreperesentable](https://khalilstemmler.com/articles/typescript-domain-driven-design/make-illegal-states-unrepresentable/#Make-illegal-states-unrepresentable).  But I guess it's actually irrelevant for the lesson we want to teach with this kata.